src/caikit_nlp_client/http_client.py

```diff
@@ -1,42 +1,19 @@
 import logging
-from dataclasses import dataclass
-from typing import Any, Optional
+from typing import Any
 
 import requests
 
 log = logging.getLogger(__name__)
 
 
-@dataclass
-class HttpConfig:
-    host: str
-    port: int
-    tls: bool = False
-    mtls: bool = False
-    client_key_path: Optional[str] = None
-    client_crt_path: Optional[str] = None
-    ca_crt_path: Optional[str] = None
-
-    def __post_init__(self):
-        if self.mtls and not self.client_key_path:
-            raise ValueError("must provide a client_key_path with mTLS")
-        if self.mtls and not self.client_crt_path:
-            raise ValueError("must provide a client_crt_path with mTLS")
-        if self.mtls and not self.ca_crt_path:
-            raise ValueError("must provide a ca_crt_path with mTLS")
-
-        if self.mtls and self.tls:
-            raise ValueError("mTLS and TLS are mutually exclusive")
-
-
 class HttpClient:
     """HTTP client for a caikit nlp runtime server
 
     Args:
         http_config (HttpConfig): Configurations to make HTTP call.
     """
 
-    def __init__(self, http_config: HttpConfig):
+    def __init__(self, host: str, port: int, secure: bool, **kwargs):
```

Comment:
Looking at it again, `secure` is actually something we only care about internally, so we could provide:


```suggestion
    def __init__(self, host: str, port: int, ca_cert: Optional[str] = None, client_cert_path: Optional[str] = None, client_key_path: Optional[str] = None):
```
then it is only up to the implementation to check what kwargs provide.

```python
if any(self._client_cert_path and not self._client_key_path, self._client_key_path and not self._client_cert_path):
    raise ValueError("for mtls, both client_cert_path and client_key_path must be provided")

if self._client_crt_path and self.client_key_path:
    # add request kwargs
    ...
```

Reply:
TLS has sligtly different requirement, I think is best then to have the user provide the scheme via a base URL argument which can be:
* `http://hostname:<optional port` for unsecured http
* `https://hostname:<optional port` for TLS and mtls.
The user would have to supply all the cert files for the client to work and be in an mTLS mode.

Reply:
Fixed in [02a508f](https://github.com/opendatahub-io/caikit-nlp-client/pull/37/commits/02a508f24b1d919217f64f05601b5892dcfd303f)